src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_nchw_kcyx_nkhw.cpp

```diff
@@ -0,0 +1,815 @@
+
+    if((in_left_pad_h > 0 || in_right_pad_w > 0 || in_left_pad_w > 0 || in_right_pad_h > 0))
+    {
+        if((GemmMPerBlock * GemmNPerBlock == 128 * 128 &&
+            GemmMPerWave * GemmNPerWave >= 128 * 64) ||
+           (GemmMPerBlock * GemmNPerBlock == 256 * 128 &&
+            GemmMPerWave * GemmNPerWave >= 128 * 64) ||
+           (GemmMPerBlock * GemmNPerBlock == 256 * 256 && GemmMPerWave * GemmNPerWave >= 128 * 128))
+        {
+            return false;
+        }
+    }
+#endif
+
+#if 0 // reduce tuning range
+    if(GemmMPerWave * GemmNPerWave != 128 * 64 || GemmMPerBlock * GemmNPerBlock != 256 * 128)
+        return false;
+#endif
+
+    if(!IsValidValue())
+        return false;
+
+    if(!IsValidXdlopsGemm_v2(
+           ctx, GemmMPerBlock, GemmNPerBlock, GemmMPerWave, GemmNPerWave, GemmKPack))
+        return false;
+
+    bool valid = false;
+
+    // check blockwise GEMM size
+    const auto n  = ConvolutionContextInterpreter::GetBatchN(ctx);
+    const auto k  = ConvolutionContextInterpreter::GetOutputChannelK(ctx);
+    const auto c  = ConvolutionContextInterpreter::GetInputChannelC(ctx);
+    const auto ho = ConvolutionContextInterpreter::GetOutputHeightHo(ctx);
+    const auto wo = ConvolutionContextInterpreter::GetOutputWidthWo(ctx);
+    const auto y  = ConvolutionContextInterpreter::GetFilterHeightY(ctx);
+    const auto x  = ConvolutionContextInterpreter::GetFilterWidthX(ctx);
+
+    const auto gemm_m       = k;
+    const auto gemm_n       = static_cast<std::size_t>(n) * ho * wo;
+    const auto gemm_k_total = static_cast<std::size_t>(c) * y * x;
+
+    if(gemm_k_total % (GemmG * GemmKPack) != 0)
+        return false;
+
+    const auto gemm_k = gemm_k_total / (GemmG * GemmKPack);
+
+    if(!(gemm_m % GemmMPerBlock == 0 && gemm_n % GemmNPerBlock == 0 && gemm_k % GemmKPerBlock == 0))
+        return false;
+
+    // check blockwise copy of A matrix
+    std::tie(std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, valid) =
+        CalculateGemmABlockCopyPerformanceParameters(ctx);
+
+    if(!valid)
+        return false;
+
+    // check blockwise copy of B matrix
+    std::tie(std::ignore, std::ignore, std::ignore, std::ignore, std::ignore, valid) =
+        CalculateGemmBBlockCopyPerformanceParameters(ctx);
+
+    if(!valid)
+        return false;
+
+    // check LDS allocation
+    std::size_t lds_size = 0;
+    std::tie(lds_size, valid) = CalculateLdsNumberOfByte(ctx);
+
+    return (valid and lds_size <= get_lds_max_number_of_byte());
+}
+
+PerformanceImplicitGemmForwardV4R4Xdlops
+ConvHipImplicitGemmForwardV4R4Xdlops::GetPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    return GetPerformanceConfigBase<PerformanceImplicitGemmForwardV4R4Xdlops>(ctx);
+}
+
+ConvSolution ConvHipImplicitGemmForwardV4R4Xdlops::GetSolution(
+    const ConvolutionContext& ctx,
+    const PerformanceImplicitGemmForwardV4R4Xdlops& config,
+    bool) const
+{
+#if 0
```

Comment:
Please remove these commented code when you are ready to merge.

Reply:
fixed